syntax = "proto3";

package kvstore;

message PutRequest {
  bytes key = 1;
  bytes value = 2;
}

message PutReply {
  // Empty because success/failure is signaled via GRPC status.
}

message GetRequest {
  bytes key = 1;
}

message GetReply {
  bytes value = 1;
}

message RemoveRequest {
  bytes key = 1;
}

message RemoveReply {
  // Empty because success/failure is signaled via GRPC status.
}

service KeyValueStore {
  rpc put (PutRequest) returns (PutReply) {}
  rpc get (stream GetRequest) returns (stream GetReply) {}
  rpc remove (RemoveRequest) returns (RemoveReply) {}
}
